Week1/homework/index.js

```diff
@@ -0,0 +1,29 @@
+'use-strict';
+
+const database = require('./databaseClasses').database;
+const country = require('./databaseClasses').country;
+const city = require('./databaseClasses').city;
+const queries = require('./query').queries;
+const questions = require('./query').questions;
+const result = require('./query').result;
+
+class Start {
+  async action() {
+    try {
+      await database.createDatabase('world');
+      await country.createCountriesTable();
+      await country.insertCountries();
+      await city.createCitiesTable();
+      await city.insertCities();
+
+      queries.forEach((query, index) => {
+        result.answer(query, questions[index]);
+      });
+    } catch (e) {
+      console.log(e.stack);
+    }
+  }
```

Comment:
The connection / pool is not closed at the end. I'm not familiar with node js but if you could find a way to release the connection at the end ?

Reply:
As you know, the advantage of using pool connection instead of using single connection is that we can run queries in parallel and it enables us to run multiple queries in a short time. Normally, when you run a NodeJS web application, you never think about shutting down your app because it should run forever but in our case after run multiple queries we have to close the database connection. The side effect of the using pool connection method of mysql module is that it doesnt have graceful end() feature to shutdown the connection. I spent 2 days just implement why it is like that, so many people are complaining about that and they suggest graceful end to overcome this problem by using mysql module.

When you use single connection and run queries then connection.end() method waits the return results of the queued queries and it closes the connection, whereas pool.end() method doesnt wait for the queries result, it just close the connection and you will get an error ''Pool is closed.." Yes there must be a way to get rid of this problem in JS world, but in MYSQL package there is not a built in way.

Reply:
What about something like :
```suggestion
// create a list of promises
     const promises = queries.map((query, index) =>  result.answer(query, questions[index]));
//wait them all to fulfill
      await Promise.all(promises);
// close the pool
       pool.end();

  }
```
in index.js ?

Reply:
Wow. Thank you for the your crystal clear solution. It works perfectly. :smile:

Reply:
I read this to find the solution :
https://medium.com/@patarkf/synchronize-your-asynchronous-code-using-javascripts-async-await-5f3fa5b1366d